streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -34,17 +34,18 @@
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-class CachingKeyValueStore<K, V> extends WrappedStateStore.AbstractStateStore implements KeyValueStore<Bytes, byte[]>, CachedStateStore<K, V> {
+class CachingKeyValueStore<K, V> extends WrappedStateStore.AbstractStateStore
+    implements KeyValueStore<Bytes, byte[]>, CachedStateStore<K, V> {
 
-    private final KeyValueStore<Bytes, byte[]> underlying;
+    final KeyValueStore<Bytes, byte[]> underlying;
     private final Serde<K> keySerde;
     private final Serde<V> valueSerde;
     private CacheFlushListener<K, V> flushListener;
     private boolean sendOldValues;
     private String cacheName;
     private ThreadCache cache;
     private InternalProcessorContext context;
-    private StateSerdes<K, V> serdes;
+    StateSerdes<K, V> serdes;
```

Comment:
It looks like this has become package-protected so that it can be accessed by subclasses, but because basically all the stores code is in one package, it has the effect of making this field visible and modifiable (since it's not final) to the entire stores module.

It would be better at least to keep the field private and add a package-protected getter.